src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -192,15 +204,11 @@ public E borrowItemWithTimeout(long timeout, TimeUnit unit) throws
 
     @Override
     public void releaseItem(E e) {
-        releaseItem(e, true);
-    }
-
-    @Override
-    public void releaseItem(E e, boolean returnToPool) {
         final ReentrantLock lock = this.queueLock;
         lock.lock();
         try {
-            if (returnToPool) {
+            // Do nothing if the item being returned is the poisonPill
```

Comment:
Another option here might be to see if the item being returned is something different than the poison pill or one of the previously registered pool instances and, if not, throw an `IllegalArgumentException` or something.  Seems like returning something that was never in the pool in the first place should be considered invalid.  That said, though, not sure if the extra validation is worth it.  I think I'm good with this as it is.  WDYT, @KevinCorcoran?

> We'll have to decide how should we handle two successive calls to .insertPill

For that, I think my vote would be to keep the first pill and not replace it with a subsequent one.  For purposes of returning the pill to a caller when translating to an error for logging, I think it would be good to hold onto information pertaining to why the original shutdown was triggered.  One could make an argument for having `.insertPill` throw an exception if a second or further pill is inserted if we're not going to retain it.  I'd probably avoid doing that, though.  In a more sophisticated solution later we might decide to hold onto an ordered list of all of the pills that were inserted for troubleshooting/error/logging purposes. I think having `.insertPill` calls beyond the first essentially be a no-op would be okay and maybe give us some flexibility for later.